library/std/src/thread/mod.rs

```diff
@@ -684,9 +684,14 @@ thread_local! {
 
 /// Sets the thread handle for the current thread.
 ///
-/// Panics if the handle has been set already or when called from a TLS destructor.
+/// Aborts if the handle has been set already to reduce code size.
 pub(crate) fn set_current(thread: Thread) {
-    CURRENT.with(|current| current.set(thread).unwrap());
+    // Using `unwrap` here can add ~3kB to the binary size. We have complete
+    // control over where this is called, so just abort if there is a bug.
+    CURRENT.with(|current| match current.set(thread) {
+        Ok(()) => {}
+        Err(_) => rtabort!("should only be set once"),
```

Comment:
This doesn't print the source location I think. So would be good to make the error give a bit of context.
```suggestion
        Err(_) => rtabort!("thread::set_current: should only be set once"),
```